### PR TITLE
[Docs] Docs update for Pipeline Parallel

### PR DESCRIPTION
**Pull request by andoorve**
This adds docs for pipeline parallel.

cc: @simon-mo @youkaichao @njhill

### CODE REVIEW

**Comment by youkaichao** [comment]
when can multiprocessing backend come in?

**Comment by andoorve** [comment]
@youkaichao as soon as #6130 goes in

File: docs/source/serving/distributed_serving.rst [hunk 1]
```diff
@@ -21,7 +21,20 @@ To run multi-GPU serving, pass in the :code:`--tensor-parallel-size` argument wh
 
     $ python -m vllm.entrypoints.openai.api_server \
     $     --model facebook/opt-13b \
-    $     --tensor-parallel-size 4
+    $     --tensor-parallel-size 4 \
```
Comment (simon-mo) [inline docs/source/serving/distributed_serving.rst:24]:
```suggestion
    $     --tensor-parallel-size 4
```

File: docs/source/serving/distributed_serving.rst [hunk 2]
```diff
@@ -21,7 +21,20 @@ To run multi-GPU serving, pass in the :code:`--tensor-parallel-size` argument wh
...
+
+You can also additionally specify :code:`--pipeline-parallel-size` to enable pipeline parallelism. For example, to run API server on 8 GPUs with pipeline parallelism and tensor parallelism:
+
+.. code-block:: console
+
+    $ python -m vllm.entrypoints.openai.api_server \
+    $     --model gpt2 \
+    $     --tensor-parallel-size 4 \
+    $     --pipeline-parallel-size 2 \
+    $     --distributed-executor-backend ray \
```
Comment (simon-mo) [inline docs/source/serving/distributed_serving.rst:34]:
```suggestion
    $     --distributed-executor-backend ray
```

**Review by simon-mo — APPROVED** [review]

**Comment by andoorve** [comment]
@simon-mo Ready for merge!

**Pull request merged**